### DISCUSSION TITLE
Avoid removing empty `dependencies`

### DISCUSSION THREAD

**Discussion by brettz9** [pnpm/pnpm#3230] [Ideas]
This is not really a bug per se, but I'd like `pnpm i` not to remove an empty `package.json` `dependencies: {}`. I believe npm used to do this but not anymore.

I like to keep it there to:

1. Satisfy package.json linter
2. allow for quick searching through large `package.json`
3. clear visual signal that there are no dependencies

Thanks!

**Comment by zkochan** [comment]
How will pnpm know to keep the empty dependencies when running `pnpm rm pkg`?

I think we need some setting for this. Something like:

```
kept-empty-manifest-fields[]=dependencies
kept-empty-manifest-fields[]=devDependencies
```

**Comment by brettz9** [comment]
Any chance you might be free to revisit this?
